src/main/java/it/gov/pagopa/admissibility/service/InitiativeStatusServiceImpl.java

```diff
@@ -0,0 +1,52 @@
+package it.gov.pagopa.admissibility.service;
+
+import it.gov.pagopa.admissibility.dto.onboarding.InitiativeStatusDTO;
+import it.gov.pagopa.admissibility.model.InitiativeConfig;
+import it.gov.pagopa.admissibility.repository.DroolsRuleRepository;
+import it.gov.pagopa.admissibility.repository.InitiativeCountersRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.Mono;
+
+import java.math.BigDecimal;
+
+@Service
+@Slf4j
+public class InitiativeStatusServiceImpl implements InitiativeStatusService {
+
+    private final DroolsRuleRepository droolsRuleRepository;
+    private final InitiativeCountersRepository initiativeCountersRepository;
+
+    public InitiativeStatusServiceImpl(DroolsRuleRepository droolsRuleRepository, InitiativeCountersRepository initiativeCountersRepository) {
+        this.droolsRuleRepository = droolsRuleRepository;
+        this.initiativeCountersRepository = initiativeCountersRepository;
+    }
+
+    @Override
+    public Mono<InitiativeStatusDTO> getInitiativeStatusAndBudgetAvailable(String initiativeId) {
+        InitiativeStatusDTO initiativeStatus = new InitiativeStatusDTO();
```

Comment:
this is wrong, you should build and then return the object inside the Mono pipeline itself, not outside, filling next its fields!

Reply:
even if it would work, it's not conceptually right, bad to read and a little less efficient